### ISSUE TITLE
Transform LookAt Target

### ISSUE THREAD

**Issue by domske** [stride3d/stride#173]
New feature / improvement: 
Add `Entity.Transform.LookAt(TargetEntity)` method. 
Let's look an object like a camera to another object.
There are some workarounds / alternatives but a game engine should have simple methods for this.
Also `Quaternion.LookRotation` would be great.
(Maybe copy such functions from other engines?)

**Comment by tebjan** [comment]
i think what you call `Quaternion.LookRotation` is `Quaternion.BetweenDirections` in Xenko.

**Comment by domske** [comment]
@tebjan I mean the Vector2 direction without roll. But right, `Entity.Transform.Rotation = Quaternion.BetweenDirections(Vector3.UnitZ, Entity.Transform.Position - Target.Transform.Position);` will work in some cases and we could reset the roll-value to `0`. 
For a camera you only want the Yaw and Pitch angle. Here is a alternative function. 

```cs
private Vector2 GetLookAtAngle(Vector3 source, Vector3 destination) {
  Vector3 dist = source - destination;
  float altitude = (float)Math.Atan2(dist.Y, Math.Sqrt(dist.X * dist.X + dist.Z * dist.Z));
  float azimuth = (float)Math.Atan2(dist.X, dist.Z);
  return new Vector2(altitude, azimuth);
}
```

Use it e.g.

```cs
Vector2 lookAngle = GetLookAtAngle(Entity.Transform.Position, Target.Transform.Position);
Quaternion result = Quaternion.RotationYawPitchRoll(lookAngle.Y, -lookAngle.X, 0);
Entity.Transform.Rotation = result;
```

The `lookAt` method should only look to the target without affecting the roll-value.
In my opinion such "shotcut" methods like `lookAt` should have a game engine.
It simplifies a lot and saves code. I think every developer who wants a camera look at a target entity will simply use this method and don't want to despair on quaternion math. 🙂

**Comment by tebjan** [comment]
the LookAt methods are defined in Matrix.cs: https://github.com/xenko3d/xenko/blob/master/sources/core/Xenko.Core.Mathematics/Matrix.cs#L1982

you should favor matrix or quaternion implementations because they avoid working with angles and the expensive trigonometry calls like Atan2.

SharpDX also has this defined for Quaternion, which might be a bit more handy here. btw, we should update the Xenko mathematics code to the latest SharpDX state, from what it was forked from...

**Comment by dfkeenan** [comment]
I have one in the XenkoToolkit, with several overloads.  

``` C#
public static void LookRotation(ref Vector3 eye, ref Vector3 target, ref Vector3 up, out Quaternion result)
```

If I put this in the engine source I would put it in the `Quaterion` struct though.  

I also created a bunch of `LookAt` extension methods for the `TransformComponent`.

**Comment by Eideren** [comment]
Implemented through #1121
